src/ZkFold/Symbolic/Compiler/ArithmeticCircuit/Instance.hs

```diff
@@ -27,13 +26,13 @@ import           ZkFold.Symbolic.Data.FieldElement                      (FieldEl
 
 ------------------------------------- Instances -------------------------------------
 
-instance (Arithmetic a, Arbitrary a) => Arbitrary (ArithmeticCircuit a Par1) where
+instance (Arithmetic a, Arbitrary a, Arbitrary (Rep i), Haskell.Ord (Rep i), Representable i, Haskell.Foldable i) => Arbitrary (ArithmeticCircuit a i Par1) where
     arbitrary = do
-        k <- integerToNatural <$> chooseInteger (2, 10)
-        let ac = mempty { acInput = [1..k], acOutput = pure k }
+        outVar <- InVar <$> arbitrary
```

Comment:
With `i ~ Vector n`, this would cause problems because the generated `outVar` most probably would be out of bounds

My proposed fix would be to change the `Rep (Vector n)` to `Zp n`

Reply:
I made this change.